content/en/tracing/guide/add_span_md_and_graph_it.md

```diff
@@ -0,0 +1,287 @@
+
+## Instrument your code with custom metadata
+
+1. **Follow the example to get your code instrumented**.
+
+    Depending on the language you are you using, you’ll need to set the metadata to add to your spans differently.
+
+    **Note**: take note of the service and [resource names][2] you are working on, these will come in handy later. In this example the service is the Ruby server `web-store` and the resource (endpoint) is `ShoppingCartController#checkout`.
+
+{{< tabs >}}
+{{% tab "Java" %}}
+
+The Datadog UI uses tags to set span level metadata. A full list of these metadata can be found in the [Datadog][1] and [OpenTracing][2] APIs.
+
+Custom metadata may be set for auto-instrumentation by grabbing the active span out of the global tracer and setting a tag with `setTag`.
+
+```java
+import io.opentracing.Tracer;
+import io.opentracing.util.GlobalTracer;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet
+class ServletImpl extends AbstractHttpServlet {
+  @Override
+  void doGet(HttpServletRequest req, HttpServletResponse resp) {
+    final Span span = GlobalTracer.get().activeSpan();
+    if (span != null) {
+      span.setTag("customer.id", 12345);
+      span.setTag("http.url", "/login");
+    }
+    // servlet impl
+  }
+}
+```
+
+[1]: https://github.com/DataDog/dd-trace-java/blob/master/dd-trace-api/src/main/java/datadog/trace/api/DDTags.java
+[2]: https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/tag/Tags.java
+{{% /tab %}}
+{{% tab "Python" %}}
+
+Add metadata directly to a span by calling `set_tag`. For example, with the following route handler:
+
+```python
+from ddtrace import tracer
+
+@app.route('/customer/<int:customer_id>')
+def handle_customer(customer_id):
+  with tracer.trace('web.request') as span:
+    span.set_tag('customer.id', customer_id)
+```
+
+The current span can be retrieved from the context in order to set its metadata. This way, if a span was started by the instrumentation, you can retrieve the span and add custom metadata. **Note**: If a span does not exist, `None` is returned:
+
+```python
+from ddtrace import tracer
+
+@app.route('/customer/<int:customer_id>')
+@tracer.wrap()
+def handle_customer(customer_id):
+  # get the active span in the context, put there by tracer.wrap()
+  current_span = tracer.current_span()
+  if current_span:
+    current_span.set_tag('customer.id', customer_id)
+```
+
+{{% /tab %}}
+{{% tab "Ruby" %}}
+
+Add metadata directly to `Datadog::Span` objects by calling `#set_tag`:
+
+```ruby
```

Comment:
If you agree with the previous comment, we can remove this section too.

Reply:
Agreed. I am also modifying the second Ruby block to call out `client_id` and not <> brackets for consistency.